### PR TITLE
Error checking for valid symbol

### PR DESCRIPTION
**Pull request by Chudleyj**
In reference to "Add error handler when stock symbol is invalid", Issue #20. 
I added the functions to check for a valid symbol. I do not know where exactly you want to make the call to check for valid symbols, so I left that out.

### CODE REVIEW

**Comment by rodolfobandeira** [comment]
Hi @Chudleyj 

First of all nice to meet you! Its great to see you here! Thanks for sharing this code and help out. 
There are some points that raised my concern:

- This will require one more dependency to the project. Not a big deal but on linux I had to install it using: `sudo apt-get install libboost-dev`. Since the method is not implemented yet, should we do this?
- When we implement these methods in the future, it will double the request time since two calls will be made. One to validate the symbol, and another one to make the real desired call. We could cache the first symbol validation on a temporary file but I was thinking on another solution.

Create a generic error handler catching bad HTTP responses returned from `libcurl`. This way, everything that is not `200 OK` would go through this error handler.

The HTTP response for the following URL is `404`, there is also a short description explaining what happened: `"Unknown symbol"`

https://api.iextrading.com/1.0/stock/invalid_symbol/price

That said, what do you guys think? @Chudleyj @mpherg 

Thanks again!
Cheers!

**Comment by rodolfobandeira** [comment]
We could also use this method in two more situations:

- Add a kind of extra flag like `--also_validate_symbol`
- Or even create another method to retrieve all symbols using this endpoint call. I believe would be really valuable to export the entire list of symbols. People could use this to do some data analysis once they have all the symbols.

File: IEX.h [hunk 1]
```diff
@@ -94,6 +96,16 @@ namespace IEX {
             out->append(in, totalBytes);
             return totalBytes;
         }
+        
+        static void parseSymbolData(const Json::Value &json_data, std::vector<std::string> &symbolVec){
+            int i = 0;
+            
+            //Step through JSON file until the end is reached
+            while(i < json_data.size()) {
```
Comment (mpherg) [inline IEX.h:104]: To remain consistent with the rest of this file consider using range-based for loop:

    for (const auto& item : json_data) {
        symbolVec.push_back(item["symbol"].asString());

**Comment by mpherg** [comment]
@Chudleyj make sure you clang-format.

@rodolfobandeira I'm with you -- I don't like making the request twice, and I like your idea about checking the HTTP return code.

File: main.cpp [hunk 2]
```diff
@@ -2,6 +2,21 @@
...
 using namespace std;
 
+
+
+
+
+bool isValidSymbol(const std::string &symbol){
+    std::vector<std::string> symbolList = IEX::Stock::getSymbolList();
+    std::string symbolCopy = symbol;
+    boost::to_upper(symbolCopy);
```
Comment (mpherg) [inline main.cpp:12]: I don't think it's worth bringing in all of Boost just for this. Here's a C++11 way of doing the same:

    for (auto &c : symbolCopy) {
      c = toupper(c);
    }

**Comment by rodolfobandeira** [comment]
Closing by inactivity

**Pull request closed without merging**